### ISSUE TITLE
bbb-html5 tries to access mongod before mongod finishes starting

### ISSUE THREAD

**Issue by ffdixon** [bigbluebutton/bigbluebutton#5193]
This problem doesn't occur on a new install, but on reboot, 

 bbb-conf --status
red5: active
tomcat7: active
nginx: active
freeswitch: active
redis-server.service: active
bbb-apps-akka: active
bbb-transcode-akka: active
bbb-fsesl-akka: active
mongod: active
**bbb-html5: failed**
bbb-webrtc-sfu: active

When looking closer

~~~
# journalctl -u bbb-html5
-- Logs begin at Thu 2018-03-01 21:39:45 UTC, end at Thu 2018-03-01 21:40:17 UTC. --
Mar 01 21:39:47 xxx-dev2 systemd[1]: Started BigBlueButton HTML5 service.
Mar 01 21:39:47 xxx-dev2 systemd_start.sh[1390]: MongoDB shell version v3.4.13
Mar 01 21:39:47 xxx-dev2 systemd_start.sh[1390]: connecting to: mongodb://127.0.0.1:27017/html5client
Mar 01 21:39:47 xxx-dev2 systemd_start.sh[1390]: 2018-03-01T21:39:47.227+0000 W NETWORK  [thread1] Failed to connect to 127.0.0.1:27017, in(checking socket for error after poll), reason: Connection r
Mar 01 21:39:47 xxx-dev2 systemd_start.sh[1390]: 2018-03-01T21:39:47.233+0000 E QUERY    [thread1] Error: couldn't connect to server 127.0.0.1:27017, connection attempt 
~~~

Need to ensure that bbb-html5 accesses mongod *after* it has finished starting up.  We've already done some work in this area, but wanted to open a ticket.

**Comment by ffdixon** [comment]
We've added the following `ExecStartPre` to `/usr/lib/systemd/system/bbb-html5.service` to wait until mongod is running before starting the HTML5 client.

~~~
[Service]
Type=simple
ExecStartPre=/bin/sh -c 'while ! /usr/bin/mongo --eval "db.version()" > /dev/null 2>&1; do sleep 0.1; done'
ExecStart=/usr/share/meteor/bundle/systemd_start.sh
~~~
This will be in the next update to the packaging .

**Comment by antobinary** [comment]
Resolved
